docs/kubebuilder_v0_v1_difference.md

```diff
@@ -0,0 +1,124 @@
+  make install
+  make run
+```
+  In v1 project, there is no generate command. When the resource or controller is updated, users don't need to regenerate the project.
+  
+## scaffolding difference
+  Init a project with example.com and add a resource/api with group apps, version v1, kind Hello in both v0 and v1. The scaffolded projects are as follows.
+  - v0 project
+```
+.
+├── cmd
+│   └── controller-manager
+│       └── main.go
+├── Dockerfile.controller
+├── Gopkg.lock
+├── Gopkg.toml
+├── hack
+│   ├── boilerplate.go.txt
+│   ├── doc.go
+│   ├── imports.go
+│   └── sample
+│       └── hello.yaml
+├── pkg
+│   ├── apis
+│   │   ├── apps
+│   │   │   ├── doc.go
+│   │   │   └── v1
+│   │   └── doc.go
+│   ├── client
+│   │   ├── clientset
+│   │   │   └── versioned
+│   │   ├── informers
+│   │   │   └── externalversions
+│   │   └── listers
+│   │       └── apps
+│   ├── controller
+│   │   ├── doc.go
+│   │   └── hello
+│   │       ├── controller.go
+│   │       ├── controller_test.go
+│   │       └── hello_suite_test.go
+│   ├── doc.go
+│   └── inject
+│       ├── args
+│       │   └── args.go
+│       ├── doc.go
+│       ├── inject.go
+│       └── zz_generated.kubebuilder.go
+└── vendor
+
+```
+  - v1 project
+```
+├── cmd
+│   └── manager
+│       └── main.go
+├── config
+│   ├── crds
+│   │   └── apps_v1_hello.yaml
+│   └── manager
+│       ├── apps_rolebinding_rbac.yaml
+│       ├── apps_role_rbac.yaml
+│       └── manager.yaml
+├── Dockerfile
+├── Gopkg.toml
+├── hack
+│   └── boilerplate.go.txt
+├── Makefile
+├── pkg
+│   ├── apis
+│   │   ├── addtoscheme_apps_v1.go
+│   │   ├── apis.go
+│   │   └── apps
+│   │       ├── group.go
+│   │       └── v1
+│   │           ├── doc.go
+│   │           ├── hello_types.go
+│   │           ├── hello_types_test.go
+│   │           ├── register.go
+│   │           └── v1_suite_test.go
+│   └── controller
+│       ├── add_hello.go
+│       ├── controller.go
+│       └── hello
+│           ├── hello_controller.go
+│           ├── hello_controller_suite_test.go
+│           └── hello_controller_test.go
+├── PROJECT
+└── vendor
+```
+Compared with v0 project, there is no `client`, `inject` folders.
+  
+## library difference
+  
+  - v0 projects import the libraries from kubebuilder, for example kubebuilder/pkg/controller. It provides a `GenericController` type with a list of functions. Note that for created resources, the corresponding client library is generated under the project folder `pkg/client`
+  
+  - v1 projects import the libraries from controller-runtime, for example controller-runtime/pkg/controller, controller-runtime/pkg/client, controller-runtime/pkg/reconcile. Note that for created resources or core types, the client library is provided by controller-runtime.
+  
+## wiring difference
+  - v0 projects has a `inject` package and it provides functions for adding the controller to controller-manager as well as registering CRDs.
```

Comment:
May be explain what wiring means here ? 
Wiring refers to the mechanics of integrating controllers and injecting the dependencies in them.

Reply:
Added